### PR TITLE
Camera timetable block

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Extracted abstract superclass from timetableBlock with subclasses CameraTimetableBlock and DirectorTimetableBlock. This means both of these blocks can have different properties (like shotblock) and different styling. The code in both classes if basically the same for now (only horizontal dragging disabled in DirectorTimetableBlock), but this will change when the DirectorTimeline and its corresponding blocks are implemented

### CODE REVIEW

File: src/main/java/gui/centerarea/CameraTimetableBlock.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package gui.centerarea;
+
+import gui.root.RootCenterArea;
+
+/**
```
Comment (Markvanderuit) [inline src/main/java/gui/centerarea/CameraTimetableBlock.java:5]: Please always add a class description. :)

File: src/main/java/gui/centerarea/CameraTimetableBlock.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+        // Mouse event handlers
+        super.addMouseEventHandlers(true);
+    }
+
+    @Override
+    void initNormalPane() {
+        // Add width/height properties, title label, count label and description label
+        super.initNormalPane();
+
```
Comment (Ezlo123) [inline src/main/java/gui/centerarea/CameraTimetableBlock.java]: Double whitespace

File: src/main/java/gui/centerarea/CameraTimetableBlock.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        this.getContentPane().getStyleClass().add("block_Foreground_Normal");
+
+    }
+
+    @Override
+    void initDraggedPane() {
+        // Init the dragpane with blur, title, count and description labels
+        super.initDraggedPane();
+
```
Comment (Ezlo123) [inline src/main/java/gui/centerarea/CameraTimetableBlock.java]: Same

File: src/main/java/gui/centerarea/DirectorTimetableBlock.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package gui.centerarea;
+
+import gui.root.RootCenterArea;
+
+/**
+ * Created by Bart.
```
Comment (Markvanderuit) [inline src/main/java/gui/centerarea/DirectorTimetableBlock.java]: Class descr.

File: src/main/java/gui/centerarea/ShotBlock.java [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
 import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import static javafx.scene.input.KeyCode.T;
 
 /**
  * Created by Bart.
```
Comment (Markvanderuit) [inline src/main/java/gui/centerarea/ShotBlock.java]: Same

File: src/main/java/gui/centerarea/TimetableBlock.java [hunk 6]
```diff
@@ -53,7 +55,7 @@
      *  Temporary variables, until they can be moved to CSS files properly.
```
Comment (Markvanderuit) [inline src/main/java/gui/centerarea/TimetableBlock.java]: Can you remove this in this PR? I forgot to do it in a previous build.

**Comment by bart-de-jonge** [comment]
Do not merge this PR yet, there is a bug present

**Comment by bart-de-jonge** [comment]
Bug is more serious than expected, closing this PR for now

File: src/main/java/gui/centerarea/ShotBlock.java [hunk 7]
```diff
@@ -1,15 +1,30 @@
...
+import javafx.scene.layout.GridPane;
 import lombok.Getter;
 import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import sun.java2d.pipe.SpanIterator;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import static javafx.scene.input.KeyCode.T;
```
Comment (martijnjanssen) [inline src/main/java/gui/centerarea/ShotBlock.java]: This import is not needed, maybe?

**Pull request merged**